pkg/dmn/dmn_engine_test.go

```diff
@@ -0,0 +1,111 @@
+package dmn_test
+
+import (
+	"errors"
+	"github.com/pbinitiative/zenbpm/pkg/dmn"
+	"gopkg.in/yaml.v3"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+type testCase struct {
+	Decision       string                 `yaml:"decision"`
+	Input          map[string]interface{} `yaml:"input"`
+	ExpectedOutput interface{}            `yaml:"expectedOutput"`
+}
+
+type configuration struct {
+	DMN   string
+	Tests []testCase `yaml:"tests"`
+}
+
+const BulkEvaluationTestPath = "./test-data/bulk-evaluation-test"
+
+func TestBulkEvaluation(t *testing.T) {
+	// setup
+	dmnEngine := dmn.New()
+
+	bulkTestConfigs, err := loadBulkTestConfigs()
+
+	if err != nil {
+		t.Fatalf("Failed to load bulk evaluation test: %v", err)
+	}
+
+	for _, configuration := range bulkTestConfigs {
+		dmnDefinition, loadErr := dmnEngine.LoadFromFile(BulkEvaluationTestPath + "/" + configuration.DMN)
+
+		if loadErr != nil {
+			t.Fatalf("Failed to load DMN file - %v", loadErr.Error())
+		}
+
+		for i, test := range configuration.Tests {
+			result, err := dmnEngine.EvaluateDRD(dmnDefinition, test.Decision, test.Input)
```

Comment:
May I propose to encapsulate the test-code inside this loop into a function, like this.
```go
testName := fmt.Sprintf("decision: '%s', input: %v", test.Decision, test.Input)
.Run(testName, func(t *testing.T) {
// inner loop here 
}
```
As a benefit, the test report tells you where an error happens, example output would like :

```
=== RUN   Test_EvaluateDRD
=== RUN   Test_EvaluateDRD/decision:_'location',_input:_map[weather:good]
--- PASS: Test_EvaluateDRD/decision:_'location',_input:_map[weather:good] (0.00s)
[...]
--- PASS: Test_EvaluateDRD (0.00s)
PASS

```